controller/controller.mustache

```diff
@@ -64,13 +65,22 @@ module.exports = {
   create: ({{name}}) => {
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
+    {{name}}.rev = uuid.v4();
     return r.table('{{pluralName}}').insert({{name}}, {returnChanges: true}).run()
     .then(rethinkdb.firstChange);
   },
   update: ({{name}}) => {
+    if (!{{name}}.rev) return Promise.reject(new httpError('BAD_REQUEST', 'No rev provided'));
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
-    return r.table('{{pluralName}}').update({{name}}, {returnChanges: true}).run()
+    return r.table('{{pluralName}}')
+    .get({{name}}.id)
+    .update(
+      rethinkdb.ifRevMatches({{name}}),
+      {returnChanges: true}
+    )
+    .run()
+    .then(rethinkdb.checkRevError)
```

Comment:
It does support a `.catch`, but the driver doesn't actually throw an error in response to a Rethink error. It just returns an object with a `first_error` property containing the error message. Kinda odd behaviour from the driver, but hey ho.

The thing we're saying is "Update this document, but bail if it looks different than the way we thought it did". I think that's best expressed by the update statement with the branch condition.

To me using the insert statement with the conflict function says:

"I don't know if this thing exists or not. If it doesn't, create it. If it does, but it looks different than the way I thought it might, bail. If it does exist and it looks like the way I thought it might, update it"

That's kind of a mouthful. I think it says too much.

Also, it introduces the possibility for a race condition bug.

Alice and Bob both query the Thing document.

Alice decides that it is a terrible document and deletes it.

Just after that, Bob adds some information to it and posts the Thing back up. He doesn't know that Alice has deleted it.

In the upsert (insert with conflict function) scenario, the code happily inserts Bob's document without ever telling him "Hey, Alice wanted this _gone_. Are you sure you want to recreate it and add stuff?"

While in the update scenario the code will go "Hey! Buddy! That document you're trying to modify is totally _gone_, man. What do you want to do?"